pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTLiteral.java

```diff
@@ -252,6 +258,100 @@ public boolean isSingleCharacterStringLiteral() {
     }
 
     public boolean isTextBlock() {
-        return isString && getImage().startsWith("\"\"\"");
+        return isString && getImage().startsWith(TEXTBLOCK_DELIMITER);
+    }
+
+    /**
+     * Returns the content of the text block after normalizing line endings to LF,
+     * removing incidental white space surrounding the text block and interpreting
+     * escape sequences.
+     */
+    @Experimental
+    public String getTextBlockContent() {
+        if (!isTextBlock()) {
+            throw new IllegalArgumentException("This is not a text block");
+        }
+
+        int start = determineContentStart(getImage());
```

Comment:
I think it would be nice to extract the rest of this routine into a static method. That way we can test on strings instead of having to parse a text block.